File: Chapter1_ClassAndObject/CSharp/Point2D.cs

//using System;
//using System.Collections.Generic;
//using System.Linq;
//using System.Text;
//using System.Threading.Tasks;

//namespace Chapter1_ClassAndObject_CSharp
//{

//}

class Point2D
{
    double x, y;
    public void set(double x0, double y0)
    {
        x = x0;
        y = y0;
    }

    public void move(double dx, double dy)
    {
        x += dx;
        y += dy;
    }

    public void scale(double sx, double sy)
    {
        x *= sx;
        y *= sy;
    }
} //end of the Point2D class

File: Chapter01.ClassAndObject/Java/Point2D.java

public class Point2D {
	double x, y;
	
	void set(double x0, double y0) {
		this.x = x0;
		this.y = y0;
	}
	
	void move(double dx, double dy) {
		this.x += dx;
		this.y += dy;
	}
	
	void scale(double sx, double sy) {
		this.x *= sx;
		this.y *= sy;
	}
}
